src/dialogs/mqtt-device-debug-info-dialog/dialog-mqtt-device-debug-info.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  LitElement,
+  css,
+  html,
+  CSSResult,
+  TemplateResult,
+  customElement,
+  property,
+} from "lit-element";
+import "../../components/ha-dialog";
+import "../../components/ha-switch";
+import { computeDeviceName } from "../../data/device_registry";
+import { computeStateName } from "../../common/entity/compute_state_name";
+import { haStyleDialog } from "../../resources/styles";
+// tslint:disable-next-line: no-duplicate-imports
+import { HaSwitch } from "../../components/ha-switch";
+import { HomeAssistant } from "../../types";
+import { MQTTDeviceDebugInfoDialogParams } from "./show-dialog-mqtt-device-debug-info";
+import { MQTTDeviceDebugInfo, fetchMQTTDebugInfo } from "../../data/mqtt";
+import "./mqtt-messages";
+import "./mqtt-discovery-payload";
+
+@customElement("dialog-mqtt-device-debug-info")
+class DialogMQTTDeviceDebugInfo extends LitElement {
+  public hass!: HomeAssistant;
+  @property() private _params?: MQTTDeviceDebugInfoDialogParams;
+  @property() private _debugInfo?: MQTTDeviceDebugInfo;
+  @property() private _showAsYaml: boolean = true;
+  @property() private _showDeserialized: boolean = true;
+
+  public async showDialog(
+    params: MQTTDeviceDebugInfoDialogParams
+  ): Promise<void> {
+    this._params = params;
+    fetchMQTTDebugInfo(this.hass, params.device.id).then((results) => {
+      this._debugInfo = results;
+    });
+  }
+
+  protected render(): TemplateResult {
+    if (!this._params || !this._debugInfo) {
+      return html``;
+    }
+
+    return html`
+      <ha-dialog
+        open
+        @closing=${this._close}
+        .heading="${this.hass!.localize(
+          "ui.dialogs.mqtt_device_debug_info.title",
+          "device",
+          computeDeviceName(this._params.device, this.hass)
+        )}"
+      >
+        <h4>Payload display</h4>
+        <ha-switch
+          .checked=${this._showDeserialized}
+          @change=${this._showDeserializedChanged}
+        >
+          ${this.hass!.localize(
+            "ui.dialogs.mqtt_device_debug_info.deserialize",
+            "format",
+            "YAML"
+          )}
+        </ha-switch>
+        <ha-switch
+          .checked=${this._showAsYaml}
+          @change=${this._showAsYamlChanged}
+        >
+          ${this.hass!.localize(
+            "ui.dialogs.mqtt_device_debug_info.show_as",
+            "format",
+            "YAML"
```

Comment:
No need to pass a variable when it is a const?

Reply:
Fixed